client/boot/index.js

```diff
@@ -383,15 +383,13 @@ function reduxStoreReady( reduxStore ) {
 	 * Layouts with differing React mount-points will not reconcile correctly,
 	 * so remove an existing single-tree layout by re-rendering if necessary.
 	 *
-	 * TODO (@seear): React 15's new reconciliation algo may make this unnecessary
+	 * TODO (@seear): Calypso-wide single tree rendering make this unnecessary
 	 */
 	page( '*', function( context, next ) {
-		const sectionNotIsomorphic = ! isSectionIsomorphic( context.store.getState() );
-		const previousLayoutIsSingleTree = ! isEmpty(
-			document.getElementsByClassName( 'wp-singletree-layout' )
-		);
+		const singleTreeSections = [ 'theme', 'themes' ];
+		const sectionName = getSectionName( context.store.getState() );
 
-		if ( sectionNotIsomorphic && previousLayoutIsSingleTree ) {
+		if ( includes( singleTreeSections, sectionName ) ) {
```

Comment:
> debug line below, which is now wrong. 

Well, no, it isn't wrong, but why would we want to re-render the multi-layout when we're about to render a complete single tree?